analyzer/codechecker_analyzer/analyzers/clangsa/analyzer.py

```diff
@@ -261,7 +261,7 @@ def get_analyzer_mentioned_files(self, output):
             return set()
 
         regex_for_ctu_ast_load = re.compile(
-            r"ANALYZE \(CTU loaded AST for source file\): (.*)")
+            r"CTU loaded AST file: (.*).ast")
```

Comment:
I think we must change the code after the regex match too.
The Clang7 regex matched the source file (with extension .c, .cpp, etc).
In Clang8, the regex matches the AST file, with the .ast extension.
Probably we need something like this:
```
path = match.group(1)[:-4] # strip the .ast extension (Do we always have a .ast extension?)
```

Anyway, we have a test which checks for the existence of the .c files in the failure zip: `test_ctu_failure_zip(self)`.
How could that test pass?

Reply:
We always add the `.ast` extension to the file name:
https://github.com/Ericsson/codechecker/blob/8eb31361d642ead646b3a0b54ab70ba42aa20382/analyzer/codechecker_analyzer/analyzers/clangsa/ctu_manager.py#L99-L100

I prefer extending the regex which is a bit more verbose and easier to understand I think compared to `[:4]` which will always strip the last 4 characters no matter what that is.

Actually the problem was that in the clang7 fork the `.ast` extension was removed by clang when the loaded files were printed out.

Reply:
> I prefer extending the regex which is a bit more verbose and easier to understand I think compared to `[:4]` which will always strip the last 4 characters no matter what that is.

But, actually, we always want to remove the last 4 character no matter what don't we? The extension will always be ".ast".
If we change the regex then we have to express in the regex to not match ".ast" from the end of the string which is going to be an ugly regex IMHO (I consider all regex ugly BTW). In contrast to the very simple [:-4] annotated with a comment.

Reply:
And consider the edge cases like `banana.ast.cpp.ast`

Reply:
The regex will only match and return this part `banana.ast.cpp` only.

Reply:
OK.